test/test_conversion.py

```diff
@@ -35,24 +36,90 @@ def test_html_to_text():
     """conversion: Test HTML to plain text
     """
 
-    def convert(body):
+    def to_html(body):
+        """
+        A function to simply html conversion tests
+        """
         return convert_between(NotifyFormat.HTML, NotifyFormat.TEXT, body)
 
-    assert convert("No HTML code here.") == "No HTML code here."
+    assert to_html("No HTML code here.") == "No HTML code here."
 
-    clist = convert("<ul><li>Lots and lots</li><li>of lists.</li></ul>")
+    clist = to_html("<ul><li>Lots and lots</li><li>of lists.</li></ul>")
     assert "Lots and lots" in clist
     assert "of lists." in clist
 
-    assert "To be or not to be." in convert(
+    assert "To be or not to be." in to_html(
         "<blockquote>To be or not to be.</blockquote>")
 
-    cspace = convert(
+    cspace = to_html(
         "<h2>Fancy heading</h2>"
         "<p>And a paragraph too.<br>Plus line break.</p>")
     assert "Fancy heading" in cspace
     assert "And a paragraph too.\nPlus line break." in cspace
 
-    assert convert(
+    assert to_html(
         "<style>body { font: 200%; }</style>"
         "<p>Some obnoxious text here.</p>") == "Some obnoxious text here."
+
+    assert to_html(
+        "<p>line 1</p>"
+        "<p>line 2</p>"
+        "<p>line 3</p>") == "line 1\nline 2\nline 3"
+
+    # Case sensitivity
+    assert to_html(
+        "<p>line 1</P>"
+        "<P>line 2</P>"
+        "<P>line 3</P>") == "line 1\nline 2\nline 3"
+
+    # double new lines (testing <br> and </br>)
+    assert to_html(
+        "some information<br/><br>and more information") == \
+        "some information\n\nand more information"
+
+    #
+    # Test bad tags
+    #
+
+    # first 2 entries are okay, but last will do as best as it can
+    assert to_html(
+        "<p>line 1</>"
+        "<p>line 2</gar>"
+        "<p>line 3>") == "line 1\nline 2\nline 3>"
+
+    # Make sure we ignore fields that aren't important to us
+    assert to_html(
+        "<script>ignore this</script>"
+        "<p>line 1</p>"
+        "Another line without being enclosed") == \
+        "line 1\nAnother line without being enclosed"
+
+    # Test cases when there are no new lines (we're dealing with just inline
+    # entries); an empty entry as well
+    assert to_html("<span></span<<span>test</span> "
+                   "<a href='#'>my link</a>") == \
+        "test my link"
+
+    result = to_html("<body><div>line 1 <b>bold</b></div> <a href='#'>my link</a>  <p>3rd line</body>")
+    import pdb
+    pdb.set_trace()
+    # </p> missing
+    assert to_html("<body><div>line 1 <b>bold</b></div>  "
+                   " <a href='#'>my link</a>"
+                   "<p>3rd line</body>") == \
+        "line 1 bold\nmy link\n3rd line"
```

Comment:
@YoRyan , this is the test I'm referring to. I believe this test should pass, and right now it doesn't. I was able to add a lot of other test variations to cover other edge cases (and updated the code to handle it). 

I believe the <p> should trigger a new line. I spent a good portion of today trying to get all of this all to work; but i'm taking a break for now (to many hours at this).  Maybe you could have a go?